### DISCUSSION TITLE
which backend technology is the best ?

### DISCUSSION THREAD

**Discussion by YahefuYusufu** [community/community#20168] [Other Feature Feedback, Questions, & Ideas]
i just finished js and react ,redux. 
i want to continue and learn a back end language , which one is the bast for my way ? 
hope someone can give me a valuable advice .
thanks

**Comment by edward795** [comment]
Go for 
**1.Node JS (JavaScript)**
**2.Django (Python)** 

Both are trending & has huge demand.Since you have already completed React & Redux it would complement if u go for Node JS making u a MERN Stack developer.(MERN-MongoDB,Express,React,Node)

**Reply by astianmuchui** [reply]
Node js , Django , Laravel , Rails

**Reply by YahefuYusufu** [reply]
thank you for your reply

**Comment by webAKAsh** [comment]
Bro go with Next.js and try to build projects with typescript using redux. This makes your stack super awesome and for database use mongoDB. next.js has both frontend and backend codes.

**Reply by YahefuYusufu** [reply]
thank you for your advice bro.

**Comment by Itz-Hyperz** [comment]
ExpressJS is definitely what I'd go with, it's an astonishing framework and has almost no limits when it comes to routing and such!

**Reply by YahefuYusufu** [reply]
thanks man , 
what is the Difference between Next.js and Express?

**Comment by AshotPoghosyan** [comment]
Node JS,
Because of using the same language Javascript on both sides, your source code will be cleaner and more consistent. You will be using the same naming convention, the same tools, and the same best practices.

**Reply by YahefuYusufu** [reply]
thanks man , makes sense

**Reply by AshotPoghosyan** [reply]
You're welcome, I hope It was helpful

**Comment by Ra55ZZ1** [comment]
JavaScript, PHP, Python, Java and Ruby are the known backend programming languages that most backend developers are using nowadays.

**Reply by YahefuYusufu** [reply]
thanks ,man

**Comment by jfullstackdev** [comment]
we are using Laravel for a full stack approach in my company, please do note there are no bests, they are just so famous because they are the ones developers are looking for but each has its own uses,

you might think C language is already outdated, then you came across Linux and you'll realize it's not dead, it's mostly used for System Programming while the same language will be problematic when you want to develop an app, say, a full stack web development, Laravel will do the job efficiently,

if say, you are dealing with services running alongside the server, you will be needing Java / Spring Boot, as this is one of the best for microservices but you'll find Laravel for these projects to be inefficient, yes you can insist that but again, it's not that efficient compared to the other one providing microservices

in short they have their own uses, unfortunately, others are simply dying because, first they may not be maintained anymore and second, it's not relevant nowadays

here in the Philippines, just these past months I was looking for a developer job, most of the openings are PHP / Laravel & Java / Spring Boot, so if you are targeting to get at least a job in the dev world, you want to learn those frameworks that are being posted in job postings

**Reply by YahefuYusufu** [reply]
thanks man

**Comment by azizsigar** [comment]
**django** 
easier than nodejs. almost everthing coming aut.

**Reply by YahefuYusufu** [reply]
i see , thank you for your reply

**Comment by YahefuYusufu** [comment]
i got the answer .thanks
